Nevi/wrangler-resource-provisioning

nevikashah opened the pull request:
### Summary

add resource provisioning beta


### Documentation checklist

- [x ] The [documentation style guide](https://developers.cloudflare.com/style-guide/) has been adhered to.
- [ x] If a larger change - such as adding a new page- an issue has been opened in relation to any incorrect or out of date information that this PR fixes.

penalosa commented on src/content/docs/workers/wrangler/resource-provisioning.mdx:
```diff
@@ -0,0 +1,54 @@
+    text: Beta
+---
+
+# Resource Provisioning
```
```suggestion
```

penalosa commented on src/content/docs/workers/wrangler/resource-provisioning.mdx:
````diff
@@ -0,0 +1,54 @@
+
+## Wrangler Configuration 
+You can set up your Wrangler configuration file like the following: 
+```json
````
This should be wrapped in the `WranglerConfig` component so it also provides TOML

penalosa commented on src/content/docs/workers/wrangler/resource-provisioning.mdx:
````diff
@@ -0,0 +1,54 @@
+When developing locally, you can interact with local versions of any KV, D1, R2 resources configured in your Wrangler configuration file by running:
+
+```sh
+npx wrangler dev –x-provision
````
These should use the `PackageManager` component to show how to do this with `pnpm` and `yarn` as well

irvinebroque reviewed (commented):
Think showing example output of Wrangler a bit more can help

Is there a reason this is top-level in the wrangler docs?

And conversely — is there a reason not to put on changelog?

nevikashah commented:
> Think showing example output of Wrangler a bit more can help
added
 
> Is there a reason this is top-level in the wrangler docs?
open to other thoughts on where to put it - struggled with this myself since its experimental. I wanted to have a singular home for it that I can eventually link out to in other sections. Could also go in Configurations > Bindings as a sub nav? 

> And conversely — is there a reason not to put on changelog?
added to changelog in this PR

irvinebroque commented on src/content/changelog/workers/2025-02-14-automatic-resource-creation-beta.mdx:
```diff
@@ -0,0 +1,45 @@
+date: 2025-02-14 00:00:00 UTC
+---
+
+You can now automatically create Cloudflare resources as part of your deployment! You no longer need to manually create any resources that do not yet exist on your account prior to deploying. Wrangler will automatically detect any required [KV](/kv), [D1](/d1), or [R2](/r2) resources and create them on your behalf. 
```
This is better than title

Don't need ! part

irvinebroque commented on src/content/changelog/workers/2025-02-14-automatic-resource-creation-beta.mdx:
```diff
@@ -0,0 +1,45 @@
+
+You can now automatically create Cloudflare resources as part of your deployment! You no longer need to manually create any resources that do not yet exist on your account prior to deploying. Wrangler will automatically detect any required [KV](/kv), [D1](/d1), or [R2](/r2) resources and create them on your behalf. 
+
+For more details, check out our [documentation](/workers/wrangler/automatic-resource-creation) on how to get started with automatic resource creation. 
```
Not sold it needs to be a named thing like "automatic resource creation"

Also doesn't match what is in the CLI flag that says "provision" - should call it same thing

irvinebroque commented on src/content/changelog/workers/2025-02-14-automatic-resource-creation-beta.mdx:
````diff
@@ -0,0 +1,45 @@
+Current Version ID: ...
+```
+
+Resource provisiong is currently a beta feature. If you run into issues or would like to leave feedback for the team to help us improve, please send us a message in our [Cloudflare Developers Discord](https://discord.cloudflare.com/) in the [Wrangler channel](https://discord.com/channels/595317990191398933/799437470004412476).
````
Just say it's beta — same stuff here about reporting stuff applies equality beta and non beta?

Beta already in title

irvinebroque commented on src/content/docs/workers/wrangler/automatic-resource-creation.mdx:
```diff
@@ -0,0 +1,97 @@
+---
+pcx_content_type: concept
+title: Automatic Resource Creation
+description: When you deploy a Worker that binds to a Cloudflare resource, Wrangler will automatically detect and create any resources that do not exist on your account. 
```
This page really makes sense too-level in wrangler docs?

irvinebroque commented on src/content/docs/workers/wrangler/automatic-resource-creation.mdx:
```diff
@@ -0,0 +1,97 @@
+---
+pcx_content_type: concept
+title: Automatic Resource Creation
```
See comment before about naming

irvinebroque commented on src/content/docs/workers/wrangler/automatic-resource-creation.mdx:
```diff
@@ -0,0 +1,97 @@
+With resource provisioning, if your Wrangler configuration file includes a [KV namespace](/kv/), [D1 database](/d1/), or [R2 bucket](/r2/) that does not yet exist, Wrangler will create these resources upon deployment. This is done by adding the experimental `-–x-provision` flag to your commands. 
+
+## Wrangler Configuration 
+You can set up your Wrangler configuration file like the following:
```
Think missing something here — the instead of <what> 

(Running commands first)

irvinebroque commented on src/content/docs/workers/wrangler/automatic-resource-creation.mdx:
````diff
@@ -0,0 +1,97 @@
+```
+</WranglerConfig>
+
+When using the experimental `-–x-provision` flag, the `id` field for KV and D1 and the `bucket_name` for R2 are optional in your wrangler configuration. 
````
This doesn't explain how it works though and how the resource is mapped to a specific KV namespace? If I have binding FOO declared in wrangler.toml for a KV namespace, and I login to the dash, how do I figure out what is what?

irvinebroque commented on src/content/docs/workers/wrangler/automatic-resource-creation.mdx:
```diff
@@ -0,0 +1,97 @@
+The newly created resource will be named with the format `<worker-name>-<binding-name>` by default.
+
+## Feedback 
+Resource provisiong is currently a beta feature. If you run into issues or would like to leave feedback for the team to help us improve, please send us a message in our [Cloudflare Developers Discord](https://discord.cloudflare.com/) in the [Wrangler channel](https://discord.com/channels/595317990191398933/799437470004412476).
```
```suggestion
Resource provisiong is currently in beta. Drop us a note in the [Cloudflare Developers Discord](https://discord.cloudflare.com/) in the [Wrangler channel](https://discord.com/channels/595317990191398933/799437470004412476) and tell us what you think
```

irvinebroque commented on src/content/changelog/workers/2025-02-14-automatic-resource-provisioning-beta.mdx:
```diff
@@ -1,14 +1,12 @@
 ---
-title: Automatic Resource Creation (Beta)
+title: Automatically provision Cloudflare resources when deploying a Worker with Wrangler (Beta)
```
```suggestion
title: Automatically provision storage and databases when deploying a Worker with Wrangler (Beta)
```

irvinebroque reviewed (commented):
Doesn't it need to be mentioned here?

https://developers.cloudflare.com/workers/wrangler/commands/

penalosa commented:
> Doesn't it need to be mentioned here?
> 
> https://developers.cloudflare.com/workers/wrangler/commands/

Probably once it's stable, but I think while it's experimental it make sense to have a dedicated page to point people to for testing? Once we stabilise this I imagine it'll need to be documented across the command and config pages as effectively just a feature of e.g. the `wrangler deploy` command (in the same vein as routes—it's just an assumed feature of `wrangler deploy` that Wrangler will update the routes for your Worker on deploy)

ToriLindsay commented on src/content/docs/workers/wrangler/automatic-resource-provisioning.mdx:
````diff
@@ -0,0 +1,95 @@
+```
+
+## Feedback 
+Resource provisiong is currently in beta. Drop us a note in the [Cloudflare Developers Discord](https://discord.cloudflare.com/) in the [Wrangler channel](https://discord.com/channels/595317990191398933/799437470004412476) and tell us what you think. 
````
```suggestion
Resource provisioning is currently in beta. Drop us a note in the [Cloudflare Developers Discord](https://discord.cloudflare.com/) in the [Wrangler channel](https://discord.com/channels/595317990191398933/799437470004412476) and tell us what you think. 
```
